### DISCUSSION TITLE
Float Number Recognition

### DISCUSSION THREAD

**Discussion by eesuck1** [community/community#53544] [Programming Help]
Good day
I intend to teach a neural network to recognize fractional numbers

I plan to divide the solution of the problem into the following stages:
1. Teach the model to recognize individual characters in printed and handwritten format
2. Teach her to recognize a comma or a dot
3. With the help of Canny edge detector, or any other,
find coordinates of objects, divide them into sub-images and use points 1 and 2
4. Then, by elementary arythmetic operations, convert this into a whole number

The main question is to generate the training data, do you have any datasets with printed numbers (for handwritten ones there is MNIST)?
Maybe you have seen similar tasks?
What do you think of my action plan in general, can it be done better or easier?

**Comment by sethjuarez** [comment]
Hi! There are a number of OCR solutions already available which would make this super-easy already - but as a learning task this would be excellent. The only handwritten number dataset I know of is MNIST unfortunately.

**Comment by hith3sh** [comment]
Hope I'm not late to answer to this discussion, 
So for the training data, I recommend you to search in Kaggle datasets, Kaggle is a open source platform where data-scientists gather around and do stuff ;), You can easily find the relevant dataset by a quick search in kaggle.
Also I think your action plan is great

**Reply by eesuck1** [reply]
Thank you for your reply!
